src/inmanta/env.py

```diff
@@ -696,6 +696,25 @@ def init_env(self) -> None:
                 raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path}")
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
         elif not os.path.exists(self._path_pth_file):
+            # make sure the venv hosts the same python version as the running process
+            if sys.platform != "win32":
+                # on UNIX based systems, the python version is in the path to the site packages dir:
+                if not os.path.exists(self.site_packages_dir):
+                    raise VenvActivationFailedError(
+                        msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch"
+                    )
+            else:
+                # get version as a (major, minor) tuple for the venv and the running process
+                venv_python_version = subprocess.check_output([self.python_path, "--version"]).decode("utf-8").split()[1]
+                venv_python_version = tuple(map(int, venv_python_version.split(".")))[:2]
+
+                running_process_python_version = sys.version_info[:2]
+
+                if venv_python_version != running_process_python_version:
+                    raise VenvActivationFailedError(
+                        msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch ({'.'.join(map(str,venv_python_version))} vs {'.'.join(map(str,running_process_python_version))})"  # NOQA E501
```

Comment:
The `# NOQA E501` should be removed here. This means that the line is too long, so we just need to spread the string across multiple lines.